src/main/clm_varctl.F90

```diff
@@ -205,6 +206,39 @@ module clm_varctl
 
   real(r8), public  :: o3_ppbv = 100._r8
 
+  ! number of wavelength bands used in SNICAR snow albedo calculation, cenlin
+  integer, public :: snicar_numrad_snw = 5 
+
+  ! type of downward solar radiation spectrum for SNICAR snow albedo calculation cenlin 
+  integer, public :: snicar_solarspec = 1  ! 1->mid-latitude winter;2->mid-latitude summer;3->sub-Arctic winter;
+                                           ! 4->sub-Arctic summer;5->Summit,Greenland,summer;6->High Mountain summer;
+
+  ! snow optics type using different refractive index databases in SNICAR, cenlin
+  integer, public :: snicar_snw_optics = 3   ! 1->Warren (1984);2->Warren and Brandt (2008);3->Picard et al (2016)
```

Comment:
As we do in other CTSM modules, I would suggest something like
```
select case (snicar_snw_optics_method)
case ('Warren1984')
   <code>
case ('Warren2008')
   <code>
case ('Picard2016')
   <code>
end select
```
rather than assigning arbitrary numbers to each method (in this case 1, 2, 3).

For examples, see how "soil_decomp_method" is used and/or search for "select case" and track down a different set of methods.

My comment also applies to `snicar_dust_optics` and likely other switches that I see in this file.